cleanlab/object_detection/rank.py

```diff
@@ -0,0 +1,1157 @@
+            f"Annotations has to be a list of dicts. Instead it is list of {type(annotations[0])}."
+        )
+    # check last column of results is probabilities ( < 1.)?
+    if not isinstance(results[0], np.ndarray):
+        raise ValueError(
+            f"Result has to be a list of np.ndarray. Instead it is list of {type(results[0])}."
+        )
+    if not results[0][0].shape[1] == 5:
+        raise ValueError(f"Result values have to be of format [_,_,_,_,pred_prob].")
+
+    valid_methods = ["map"]
+    if method is not None and method not in valid_methods:
+        raise ValueError(
+            f"""
+            {method} is not a valid object detection scoring method!
+            Please choose a valid scoring_method: {valid_methods}
+            """
+        )
+
+    if threshold is not None and threshold > 1.0:
+        raise ValueError(
+            f"""
+            Threshold is a cutoff of predicted probabilities and therefore should be <= 1.
+            """
+        )
+
+
+def _get_bbox_labels_annotation(annotation):
+    """Returns bbox and label values for annotation."""
+    bboxes = annotation["bboxes"]
+    labels = annotation["labels"]
+    return bboxes, labels
+
+
+def _get_bbox_labels_result(result):
+    """Returns bbox, label and pred_prob values for result."""
+    labels = []
+    boxes = []
+    for idx, result_class in enumerate(result):
+        labels.extend([idx] * len(result_class))
+        boxes.extend(result_class.tolist())
+    bboxes = [box[:4] for box in boxes]
+    pred_probs = [box[-1] for box in boxes]
+    return bboxes, labels, pred_probs
+
+
+# ==========TO BE DEPRECATED: Example score (calculate mAP)=============
+def _calculate_map(annotations, results):
+    map = np.zeros((len(annotations),))
+    for i, (d, r) in enumerate(zip(annotations, results)):
+        map[i] = bbox_map_eval(d, r)
+    return map
+
+
+def bbox_map_eval(det_result, annotation, nproc=4):
+    """Evaluate mAP of single image det result.
+    Args:
+        det_result (list[list]): [[cls1_det, cls2_det, ...], ...].
+            The outer list indicates images, and the inner list indicates
+            per-class detected bboxes.
+        annotation (dict): Ground truth annotations where keys of
+             annotations are:
+            - bboxes: numpy array of shape (n, 4)
+            - labels: numpy array of shape (n, )
+            - bboxes_ignore (optional): numpy array of shape (k, 4)
+            - labels_ignore (optional): numpy array of shape (k, )
+        nproc (int): Processes used for computing mAP.
+            Default: 4.
+    Returns:
+        float: mAP
+    """
+    # use only bbox det result
+    if isinstance(det_result, tuple):
+        bbox_det_result = [det_result[0]]
+    else:
+        bbox_det_result = [det_result]
+    # mAP
+    iou_thrs = np.linspace(0.5, 0.95, int(np.round((0.95 - 0.5) / 0.05)) + 1, endpoint=True)
+
+    processes = []
+    workers = Pool(processes=nproc)
```

Comment:
we'll stick to IOU for single threshold? and keep 1 processor?

Reply:
This is just an example function, mAP should be truncated in favor of algorithm we're developing.
But good catch I can change to correct IOU if we choose to keep mAP as an optional score.